core/factory.py

```diff
@@ -124,7 +125,14 @@ def is_member(self, uid: int, whitelist: List[int]) -> bool:
             # whitelist is a whitelist, we don't lock whitelisted users
             return True
         return False
-
+    
+    def is_blogger(self, message: Message, whitelist: list) -> bool:
+        """
+        Checks if the user is writing as a channel
+        """
+        user = message.from_user
+        return False if user.first_name == 'Channel' and user.is_bot else True
```

Comment:
I think it is vice versa: this expression yields opposite result. This should be correct:
```python
return user.first_name == 'Channel' and user.is_bot
```

Also, we should be using `whitelist`. There seems to be `sender_chat` field of Message object, it should have `id` attribute, we can use that to check against. Something like this:
```python
if message.sender_chat.id in whitelist:
  return False
return user.first_name == 'Channel' and user.is_bot
```

But for this to work properly, we need to make sure that `sender_chat` field exists in `message` object